[CIAMP-1281] add note about missing deploy/user keys (#9282)
circleci/circleci-docs: jekyll/_cci2/github-integration.adoc

To achieve fine-grained access to more than one repository, consider creating what GitHub calls a <<#controlling-access-via-a-machine-user,machine user>>. Give this user exactly the permissions your build requires, and then associate its user key with your project on CircleCI.

[NOTE]
====

> If your deploy keys or user keys appear to have gone missing, it may be due to one of the following reasons:

Reviewer: ```suggestion
If your deploy keys or user keys appear to have been removed, it may be due to one of the following reasons:
```

Author: updated 👍

> * GitHub will remove inactive keys if no one uses them for over a year.

Reviewer: ```suggestion
* GitHub will remove inactive keys if they are unused for over a year.
```

* If CircleCI creates keys through a user's OAuth integration with GitHub, and the user revokes or removes the integration, GitHub will also remove the associated keys.
* In GitHub, if an organization owner restricts or disables deploy keys across all repositories, GitHub may disable or remove existing deploy keys.

> * If your CircleCI project has no followers, CircleCI will consider it disabled and remove the associated keys.

Reviewer 2: I don't know if we might want a link here to something about followers? Not sure

Reviewer: It is explained here: https://circleci.com/docs/introduction-to-the-circleci-web-app/#projects, that's probably not ideal but likely the best place for now

Author: added a link 👍

> * If your CircleCI project has no xref:introduction-to-the-circleci-web-app/#projects[followers], CircleCI will consider it disabled and remove the associated keys.

Reviewer: ```suggestion
* If your CircleCI project has no xref:introduction-to-the-circleci-web-app#projects[followers], CircleCI will consider it disabled and remove the associated keys.
```
